rust/xaynet-mobile/src/participant.rs

```diff
@@ -0,0 +1,239 @@
+use std::{convert::TryInto, sync::Arc};
+
+use thiserror::Error;
+use tokio::{
+    runtime::Runtime,
+    sync::{mpsc, Mutex},
```

Comment:
Is there a reason for reaching for the tokio mutex here? Specifically, do we _want_ locking to be asynchronous?